src/openms/include/OpenMS/FILTERING/CALIBRATION/PrecursorCorrection.h

```diff
@@ -0,0 +1,217 @@
+     @param all_matching_features: bool correction is perfored for all features in question not only the closest one.
+     @param max_trace: integer maximal number of traces used.
+     @param debug_level: integer debug level.
+     @return set of Size with corrected precursor information.
+
+
+     */
+     static std::set<Size> correctToNearestFeature(const FeatureMap& features,
+                                                   MSExperiment & exp,
+                                                   double rt_tolerance_s = 0.0,
+                                                   double mz_tolerance = 0.0,
+                                                   bool ppm = true,
+                                                   bool believe_charge = false,
+                                                   bool keep_original = false,
+                                                   bool all_matching_features = false,
+                                                   int max_trace = 2,
+                                                   int debug_level = 0);
+
+  protected:
+
+      /**
+      @brief Check if precursor is located in the bounding box of a features convex hull.
+      Here the bounding box of the feature is extended by the retention time tolerance and
+      afterwards the precursor location is validated.
+
+      @param feature: constant Feature.
+      @param rt: constant double retention time.
+      @param pc_mz: constant double precursor mass to charge.
+      @param rt_tolerance: constant double retention time tolerance in seconds.
+      @return static boolean to check if the precursor is located in the bounding box of a features convex hull.
+      */
+      static bool overlaps_(const Feature& feature,
+                            const double rt,
+                            const double pc_mz,
+                            const double rt_tolerance);
+
+      /**
+      @brief Check precursor and feature compatiblity
+      If the precursor mz is in one of the masstraces the feature is compatible.
+      Dependend on 13C mass difference and charge.
+
+      @param feature: constatnt Feature.
+      @param pc_mz: double precursor mass to charge.
+      @param mz_tolerance: double mass to charge tolerance.
+      @param max_trace_number: Size maximum number of mass traces.
+      @param debug_level: integer debug level.
+      @return static boolean if the precursor mass to charge is in one of the features masstraces.
+      */
+      static bool compatible_(const Feature& feature,
+                              double pc_mz,
+                              double mz_tolerance,
+                              Size max_trace_number = 2,
+                              int debug_level = 0);
+};
+
+const std::string PrecursorCorrection::csv_header = "RT,uncorrectedMZ,correctedMZ,deltaMZ";
```

Comment:
/PrecursorCorrection.h(215): error C2491: 'OpenMS::PrecursorCorrection::csv_header': definition of dllimport static data member not allowed

Reply:
I think you have to *declare* this in the header and then *define* it in the cpp otherwise Windows becomes unhappy

Reply:
Thank you!